docs/how-to-add-form.md

```diff
@@ -0,0 +1,258 @@
+
+| Column | Description |
+| --- | --- |
+| submissionStatusId | Primary Key - integer |
+| submissionId | Foreign Key - from app\_submission.submissionId |
+| code | Foreign Key - from app\_status\_code.code  |
+| classification | Should be from app\_status\_code.allowedClassifications for the code. |
+| assignedTo | Name of team member assigned to this submission.  |
+| assignedEmail | Email address of team member assigned to this submission.  |
+| actionDate | Date relevant to the status (when assigned, when completed, etc. ) |
+
+#### app\_note
+Notes for a submission or for a submission status. Provide additional context about the submission as a whole, or about a particular status (ex. Notes for the team member to take after they have been assigned, or notes about the work they have done and justification for a Pass or Fail when marking Complete).  
+
+| Column | Description |
+| --- | --- |
+| noteId | Primary Key - integer |
+| submissionId | Foreign Key - from app\_submission.submissionId |
+| submissionStatusId | Foreign Key - from app\_submission\ submissionStatusId |
+| note | Free form text field. |
+
+#### app\_settings
+Runtime configuration or settings for your forms, emails, etc.  
+
+| Column | Description |
+| --- | --- |
+| name | Primary Key - string, name for the setting |
+| enabled | Boolean - is the setting currently valid and useable |
+| config | JSONB - store a json configuration in any shape you need. | 
+
+### Sample Migrations
+Sample [migrations](sample/db/migrations) illustrate adding your implementations of the 'common' tables, your form/business specific tables, and initializing the data (status codes, settings).  
+
+This assumes that the main Metadata/Form table has already been migrated to your database.
+
+```js
+const stamps = require('../stamps');
+
+exports.up = function(knex) {
+  return Promise.resolve()
+    .then(() => knex.schema.createTable('form', table => {
+      table.uuid('formId').primary();
+      table.string('name').unique().notNullable();
+      table.string('slug').unique().notNullable().comment('This will be used for paths and user interactions');
+      table.boolean('public').notNullable().defaultTo(true);
+      table.boolean('active').notNullable().defaultTo(true);
+      table.specificType('keywords', 'text ARRAY');
+      table.string('prefix').unique().notNullable().comment('This is the database table prefix for all the form business tables.');
+      stamps(knex, table);
+    }));
+};
+
+exports.down = function(knex) {
+  return Promise.resolve()
+    .then(() => knex.schema.dropTableIfExists('form'));
+};
+```
+
+## API (Node Server)
+Out of the box, the 'common' tables have [models](../app/src/forms/common/models), a [controller](../app/src/forms/common/controller.js), a [data service](../app/src/forms/common/dataService.js) and [routes](../app/src/forms/common/router.js).  Also included: [Team management](../app/src/forms/teammanagement) (see Keycloak section). 
```

Comment:
Team management link should also point to `team-management.md` in the docs folder in addition to pointing out what folder the implementation code is in (the readme was lifted out of that folder and placed into /docs in #132 )